src/qappfw.js

```diff
@@ -305,22 +310,34 @@ class QRadar {
      * @throws Error if aql is not supplied or if the search results could not be displayed.
      *
      */
+  
     static openEventSearch(aql, openWindow)
     {
         if (aql == null)
         {
             throw new Error("You must supply an AQL string");
         }
-
-        return QRadar.windowOrTab(
-            "do/ariel/arielSearch?appName=EventViewer&pageId=EventList&dispatch=performSearch" +
-            "&value(timeRangeType)=aqlTime&value(searchMode)=AQL" +
-            "&value(aql)=" + encodeURIComponent(aql), openWindow === false ? "EVENTVIEWER" : null);
+        // If qradar version is greater than 202164 then it is UP4 or above
+        if (QRadar._isUP4AndAbove() >= 202164)
```

Comment:
Could the `_isUP4AndAbove()` function return a boolean instead? So if it's UP4+ it returns `true` and if not it returns `false`? This way you could just have `if (QRadar._isUP4AndAbove())` for this if statement, which would be a bit easier to read.

Reply:
Yeah, ok no worries so do you mean something on these lines?:
 ```
static _isUP4AndAbove()
    {
        // Creates variables for fetching qradar version, removing the dots and taking the first 6 characters from it, i.e. 202164
        var fullQradarVersion = top.QRADAR_VERSION.replaceAll(".","");
        var qradarVersion = fullQradarVersion.substring(0,6);
        if (qradarVersion >= 202164) { return true; }
        else { return false; }
    }
```